Models/GenerateImageViewModel.swift

```diff
@@ -1,53 +1,58 @@
-//
-//  ModelView.swift
-//  iosdevapp
-//
-//  Created by Leonit Ejupi on 07.02.23.
-//
-
 import Foundation
+import UIKit
 
 class GenerateImageViewModel: ObservableObject {
     private let urlSession: URLSession
     @Published var imageURL: URL?
     @Published var isLoading = false
     @Published public var text: String = "Two astronauts"
-    
+    @Published var showSavedImageAlert = false
+    private var apiKey = ""
+
     init(urlSession: URLSession = URLSession.shared) {
         self.urlSession = urlSession
+
+        do {
+            let APISecretKey = try fileLoaderService(fileName: "APIKey")
+            apiKey = APISecretKey.apiKey
+        } catch {
+            print("Failed to load API key: \(error.localizedDescription)")
+        }
```

Comment:
I would move the whole request and read API key functionality to an API service.